proposals/NNNN-GetInteriorVehicleData.md

```diff
@@ -0,0 +1,96 @@
+
+
+## Proposed solution
+
+The following sequence diagram shows the proposed behavior of sdl regarding GetInteriorVehicleData. 
+
+![../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png](../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_new_subscribe.png)
+
+SDL maintains the module (Type) subscription status as a whole. SDL need to subscribe to a module if there is at least one app subscribes to the module. SDL need to un-subscribe to a module if there is no app subscribes to the module. Or equivalently
+-	SDL shall only include parameter `subscribe=true` if SDL has not (equivalently there is no app) subscribed to the module data when forwarding GetInteriorVehicleData with `subscribe=true`.
+-	SDL shall only include parameter `subscribe=false` if it is the last app to unsubscribe to the module data when forwarding GetInteriorVehicleData with `subscribe=false`.
+
+
+SDL maintains a cache of module data only when there is at least an application subscribed to the module. 
+-	SDL creates a cache for the module when it receives a successful GetInteriorVehicleData response with `isSubscribed=true`. 
+-	SDL updates the cache (only the part that changes) when it receives an OnInteriorVehicleData notification. 
+-	SDL removes the cache for the module when SDL receives a successful GetInteriorVehicleData response with `isSubscribed=false`. 
+	
+
+SDL only forwards a GetInteriorVehicleData request to HMI if there is no cached data available for the requested module or it needs to un-subscribe to the module from HMI. Otherwise, SDL responds the resquest with the cached data and no need to forward to and wait a response from HMI.
+
+In addition, when there is no cache available and before SDL forwards a GetInteriorVehicleData request to HMI, SDL shall apply a rate limitation scheme similar to that of the GetVehicleData request. SDL allows `x` number of GetVehicleData requests per second. The number `x` is 5 and it can be configured in smartDeviceLink.ini file. Similarly, SDL allows `y` number of GetInteriorVehicleData requests to forward to HMI. The number `y` shall be configurable in ini file. Since there are multiple moduleTypes defined for RC, ideally SDL shall apply rate limitation per moduleType. To simplify the implementation it can be regardless of moduleType. However, in this case, `y` shall be much greater than `x`.
+
+
+```
+; Limitation for a number of GetVehicleData requests (the 1st value) per (the 2nd value) seconds
+GetVehicleDataRequest = 5, 1
++; Limitation for a number of GetInteriorVehicleDataRequest requests (the 1st value) per (the 2nd value) seconds
++GetInteriorVehicleDataRequest = 20, 1
+```
+
+## Potential downsides
+
+- SDL core need more memory for the data cache. In case there are something wrong with the HMI subsription, SDL will not get data updated. GetInteriorVehicleData will have aged data. 
```

Comment:
Please update to "...needs more memory..." and "In the case there is something wrong..."